src/boot/efi/boot.c

```diff
@@ -2028,16 +2028,28 @@ static VOID config_entry_add_linux(
                                 os_build_id = stra_to_str(value);
                                 continue;
                         }
+
+                        if (strcmpa((const CHAR8*) "IMAGE_VERSION", key) == 0) {
+                                FreePool(os_image_version);
+                                os_image_version = stra_to_str(value);
+                                continue;
+                        }
                 }
 
-                if ((os_name_pretty || os_name) && os_id && (os_version || os_version_id || os_build_id)) {
+                if ((os_name_pretty || os_name) && os_id && (os_image_version || os_version || os_version_id || os_build_id)) {
                         _cleanup_freepool_ CHAR16 *path = NULL;
 
                         path = PoolPrint(L"\\EFI\\Linux\\%s", f->FileName);
 
-                        entry = config_entry_add_loader(config, device, LOADER_LINUX, f->FileName, 'l',
-                                                        os_name_pretty ?: os_name, path,
-                                                        os_version ?: (os_version_id ? : os_build_id));
+                        entry = config_entry_add_loader(
+                                        config,
+                                        device,
+                                        LOADER_LINUX,
+                                        f->FileName,
+                                        /* key= */ 'l',
+                                        os_name_pretty ?: os_name,
+                                        path,
+                                        os_image_version ?: (os_version ?: (os_version_id ? : os_build_id)));
```

Comment:
Would it work to add "n/a" at the end of this chain to fix #20820 ? Things wouldn't be sorted, but if one wants that then they need to provide a version

Reply:
i think the fix for that should look different, i.e. we might just want to pass NULL in that case. the version is optional for the rest of the code and used for presentation purposes only (unlike i suggested elsewhere we actually order entries by the filename where they are defined in, not the version)